src/render/portable/mod.rs

```diff
@@ -344,6 +345,42 @@ fn render_table_javascript<P: AsRef<Path>>(
         })
         .collect();
 
+    let heatmap_brush_domains: HashMap<String, Vec<f32>> = render_columns
+        .iter()
+        .filter(|(title, _)| *numeric.get(&title.to_string()).unwrap())
+        .filter(|(_, k)| k.plot.is_some())
+        .filter(|(_, k)| k.plot.as_ref().unwrap().heatmap.is_some())
+        .filter(|(_, k)| {
+            k.plot
+                .as_ref()
+                .unwrap()
+                .heatmap
+                .as_ref()
+                .unwrap()
+                .domain
+                .is_some()
+        })
+        .map(|(title, k)| {
+            (
+                title.to_string(),
+                k.plot
+                    .as_ref()
+                    .unwrap()
+                    .heatmap
+                    .as_ref()
+                    .unwrap()
+                    .domain
+                    .as_ref()
+                    .unwrap()
+                    .iter()
+                    .map(|s| f32::from_str(s).unwrap())
+                    .collect_vec(),
+            )
+        })
```

Comment:
I think this can be implemented more efficient and with less code when using `filter_map`.

Reply:
Same holds for the tickplot case above.